src/main/java/no/hvl/dat250/h2020/group5/requests/VoteRequestFromDevice.java

```diff
@@ -0,0 +1,37 @@
+package no.hvl.dat250.h2020.group5.requests;
+
+import lombok.Getter;
+import lombok.Setter;
+import no.hvl.dat250.h2020.group5.entities.Vote;
+import no.hvl.dat250.h2020.group5.enums.AnswerType;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Setter
+@Getter
+public class VoteRequestFromDevice {
+
+  private int numberOfYes = 0;
+  private int numberOfNo = 0;
+  private Long pollId;
+  private Long deviceId;
+
+  public VoteRequestFromDevice(Long deviceId, Long pollId, int numberOfYes, int numberOfNo) {
+    this.numberOfNo = numberOfNo;
+    this.numberOfYes = numberOfYes;
+    this.pollId = pollId;
+    this.deviceId = deviceId;
+  }
+
+  public List<Vote> getVotes() {
+    List<Vote> votes = new ArrayList<>();
+    for (int i = 0; i < numberOfYes; i++) {
+      votes.add(new Vote().answer(AnswerType.YES));
+    }
+    for (int i = 0; i < numberOfNo; i++) {
+      votes.add(new Vote().answer(AnswerType.NO));
+    }
+    return votes;
+  }
+}
```

Comment:
Er det slik at du ikke setter pollId her fordi de må lagres først? 🕵️

Reply:
Ja, for hvis vote.setPoll(poll) kalles på her, så kaller den videre på poll.addVote(vote), men vote har ikke fått noen ID enda siden de ikke er lagret, og da blir vote_id lik null. :)